all_test.go

```diff
@@ -77,6 +80,22 @@ func TestFetch_03(t *testing.T) {
 	Expect(t, og.Image[0].URL).ToBe("http://www-cdn.jtvnw.net/images/twitch_logo3.jpg")
 }
 
+func TestFetchWithContext(t *testing.T) {
+	s := dummySlowServer(time.Millisecond * 300)
+	defer s.Close()
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Millisecond*750)
```

Comment:
IMO
I don't think we need to use the same context for following 3 requests and 2 assertions. Using 1 context for different scenarios makes the test case confusing.